java/code/src/com/redhat/rhn/domain/action/ActionFormatter.java

```diff
@@ -213,5 +229,44 @@ public Object getDetails(Server server) {
         return getDetails(server, null);
     }
 
+    private String formatResultMessage(String msg) {
+        Yaml yaml = new Yaml();
+        List<StateResult> result = new LinkedList<>();
+        try {
+            @SuppressWarnings("unchecked")
+            Map<String, Map<String, Object>> payload = yaml.loadAs(msg, Map.class);
+            payload.entrySet().stream().forEach(e -> result.add(new StateResult(e)));
+        }
+        catch (ConstructorException ce) {
+            return StringEscapeUtils.escapeHtml4(msg);
+        }
+        return ActionFormatter.formatSaltResultMessage(result);
+    }
+
+    /**
+     * Format a list of StateResults into human readable and formatted text
+     * @param result list of state results
+     * @return human readable and formatted text
+     */
+    public static String formatSaltResultMessage(List<StateResult> result) {
+        StringBuilder retval = new StringBuilder();
+        result.stream()
+                .sorted(
+                        (r1, r2) -> (r2.getRunNum() < r1.getRunNum()) ? 1 : -1)
```

Comment:
I think it's cleaner to just use the standard `Comparator` API:

```suggestion
                .sorted(Comparator.comparingDouble(StateResult::getRunNum))
```